File: java/src/test/java/com/example/livecoding/stacksqueues/QueueUsingTwoStacksTest.java
package com.example.livecoding.stacksqueues;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class QueueUsingTwoStacksTest {

    @Test
    void fifoOrder() {
        QueueUsingTwoStacks q = new QueueUsingTwoStacks();
        q.push(1);
        q.push(2);
        assertEquals(1, q.peek());
        assertEquals(1, q.pop());
        assertEquals(2, q.peek());
    }

    @Test
    void interleaved() {
        QueueUsingTwoStacks q = new QueueUsingTwoStacks();
        q.push(1);
        assertEquals(1, q.pop());
        q.push(2);
        q.push(3);
        assertEquals(2, q.pop());
    }

    @Test
    void empty() {
        QueueUsingTwoStacks q = new QueueUsingTwoStacks();
        assertTrue(q.empty());
        q.push(1);
        assertFalse(q.empty());
        q.pop();
        assertTrue(q.empty());
    }
}

File: typescript/tests/queueUsingTwoStacks.test.ts
import { describe, expect, it } from "vitest";

import { QueueUsingTwoStacks } from "../src/stacks_queues/queueUsingTwoStacks.js";

describe("QueueUsingTwoStacks", () => {
  it("fifo", () => {
    const q = new QueueUsingTwoStacks();
    q.push(1);
    q.push(2);
    expect(q.peek()).toBe(1);
    expect(q.pop()).toBe(1);
    expect(q.peek()).toBe(2);
  });

  it("interleaved", () => {
    const q = new QueueUsingTwoStacks();
    q.push(1);
    expect(q.pop()).toBe(1);
    q.push(2);
    q.push(3);
    expect(q.pop()).toBe(2);
  });

  it("empty", () => {
    const q = new QueueUsingTwoStacks();
    expect(q.empty()).toBe(true);
    q.push(1);
    expect(q.empty()).toBe(false);
    q.pop();
    expect(q.empty()).toBe(true);
  });
});
